src/veh_interact.cpp

```diff
@@ -572,29 +584,155 @@ void veh_interact::do_install()
+                                                   part.has_flag(VPFLAG_CONTROLS) ||
+                                                   part.location == "fuel_source" ||
+                                                   part.location == "on_battery_mount" ||
+                                                   (part.location.empty() && part.has_flag("FUEL_TANK")); };
+    tab_filters[tab_filters.size()-1] = [&](vpart_info part) { // Other: everything that's not in the other filters
+        for (size_t i=1; i < tab_filters.size()-1; i++ ) {
+            if( tab_filters[i](part) ) return false;
+        }
+        return true; };
+
+    std::vector<vpart_info> tab_vparts = can_mount; // full list of mountable parts, to be filtered according to tab
+
     int pos = 0;
+    size_t tab = 0;
     while (true) {
-        sel_vpart_info = &(can_mount[pos]);
-        display_list (pos, can_mount);
+        display_list(pos, tab_vparts);
+
+        // draw tab menu
+        int tab_x = 0;
+        for( size_t i=0; i < tab_list.size(); i++ ){
+            std::string tab_name = (tab == i) ? tab_list[i] : tab_list_short[i]; // full name for selected tab
+            tab_x += (tab == i); // add a space before selected tab
+            draw_subtab(w_list, tab_x, tab_name, tab == i, false);
+            tab_x += ( 1 + utf8_width(tab_name.c_str()) + (tab == i) ); // one space padding and add a space after selected tab
+        }
+        wrefresh(w_list);
+
+        sel_vpart_info = (tab_vparts.size() > 0) ? &(tab_vparts[pos]) : NULL; // filtered list can be empty
+
+        if (sel_vpart_info != NULL ) {
+            display_details(*sel_vpart_info);
+        }
         bool can_install = can_install_part(msg_width);
-        
+
         const std::string action = main_context.handle_input();
         if (action == "INSTALL" || action == "CONFIRM"){
             if (can_install) {
-                sel_cmd = 'i';
-                return;
+                std::vector< vpart_info* > shapes = vpart_shapes[sel_vpart_info->name];
+                int selected_shape = -1;
+                if ( shapes.size() > 1 ) { // more than one shape available, display selection
+                    std::vector<uimenu_entry> shape_ui_entries;
+                    for ( size_t i = 0; i < shapes.size(); i++ ) {
+                        uimenu_entry entry = uimenu_entry( i, true, UIMENU_INVALID,
+                                                           shapes[i]->name );
+                        entry.extratxt.left = 1;
+                        entry.extratxt.sym = special_symbol( shapes[i]->sym );
+                        entry.extratxt.color = shapes[i]->color;
+                        shape_ui_entries.push_back( entry );
+                    }
+                    selected_shape = uimenu( true, getbegx(w_list), list_w, getbegy(w_list),
+                                             "Choose shape:", shape_ui_entries ).ret;
```

Comment:
Little more detail: adding `_()` , so `_("Choose shape:")` , lets our localization code grab this for translation and replacement as needed.  It's pretty important and if we know it's missing, such as here, the PR's not mergeable.